config/samples/traits/route.yaml

```diff
@@ -1,42 +1,33 @@
 apiVersion: core.oam.dev/v1alpha2
 kind: TraitDefinition
 metadata:
-  name: ingresses.networking.k8s.io
+  name: routes.standard.oam.dev
   annotations:
-    "definition.oam.dev/apiVersion": "networking.k8s.io/v1beta1"
-    "definition.oam.dev/kind": "Ingress"
+    definition.oam.dev/apiVersion: standard.oam.dev/v1alpha1
+    definition.oam.dev/kind: Route
```

Comment:
Is this annotation auto patched?

Why don't we reuse `definitionRef` (with apiVersion)?

Reply:
we can't easily get version and kind by definitionRef , so we add annotation here in vela.

Reply:
It should be fixed in `definitionRef` eventually, otherwise we should deprecate this field, for example:

```yaml
definitionRef:
    name: routes.standard.oam.dev
    version: v1alpha1
```

or 

```yaml
definitionRef:
  apiVersion: standard.oam.dev/v1alpha1
  kind: Route